### PR TITLE
Fix access token validation and revoking exception handling

### PR DESCRIPTION
**Pull request by dacrome**
- Catch `UnauthorizedException` and throw Spring `InvalidTokenException`
  which response implicit with `401 UNAUTHORIZED`

### CODE REVIEW

File: src/main/java/org/osiam/security/authorization/AccessTokenValidationService.java [hunk 1]
```diff
@@ -129,9 +133,9 @@ public void revokeAccessTokens(String id, String token) {
             LOGGER.warn(String.format("Unable to revoke access token on auth-server %s: %s",
                     authServerHome, e.getMessage()));
             throw e;
-        } catch(OsiamRequestException e) {
+        } catch (UnauthorizedException e) {
             LOGGER.warn("Error revoking all access tokens", e);
-            throw e;
+            throw new InvalidTokenException("The provided access token is not valid", e);
```
Comment (wallner) [inline src/main/java/org/osiam/security/authorization/AccessTokenValidationService.java:138]: We don't log the exception in the other log statements. I also don't think it's a good idea, it just clutters the logs for operations, we should just log the message:

`LOGGER.warn(String.format("Error revoking all access tokens on auth-server %s: %s", authServerHome,  e.getMessage()))`

File: src/main/java/org/osiam/security/authorization/AccessTokenValidationService.java [hunk 2]
```diff
@@ -143,9 +147,9 @@ private AccessToken validateAccessToken(String token) {
             LOGGER.warn(String.format("Unable to retrieve access token from auth-server %s: %s",
                     authServerHome, e.getMessage()));
             throw e;
-        } catch(OsiamRequestException e) {
+        } catch (UnauthorizedException e) {
             LOGGER.warn("Error validating access token", e);
```
Comment (wallner) [inline src/main/java/org/osiam/security/authorization/AccessTokenValidationService.java]: Even if it wasn't in this commti I think e should be consistent regarding the logging.

File: src/main/java/org/osiam/security/authorization/AccessTokenValidationService.java (hunk 1 above)
Comment (dacrome) [inline src/main/java/org/osiam/security/authorization/AccessTokenValidationService.java:138]: The auth-server endpoint is not necessary here, but I agree not to log the stack trace.

**Comment by wallner** [comment]
thank you.

**Pull request merged**